File: Data Structures & Algorithms/top-k-elements-in-list/submission-12.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq_map = {}
        for i in nums:
            freq_map[i] = freq_map.get(i, 0) + 1

        buckets = [[] for _ in range(len(nums) + 1)]
        for i, j in freq_map.items():
            buckets[j].append(i)
        
        res = []
        for i in buckets:
            if i:
                res.extend(i)
        res.reverse()

        return res[0:k]

File: Data Structures & Algorithms/trapping-rain-water/submission-12.cpp
class Solution {
public:
    int trap(vector<int>& height) {
        int l = 0, r = height.size() - 1, lm = height.at(l), rm = height.at(r);
        int res = 0;
        while (l < r) {
            if (lm < rm) {
                ++l;
                lm = max(lm, height.at(l));
                res += lm - height.at(l);
            } else {
                --r;
                rm = max(rm, height.at(r));
                res += rm - height.at(r);
            }
        }
        return res;
    }
};
